pkg/types/baremetal/validation/platform.go

```diff
@@ -109,6 +109,10 @@ func validateOSImageURI(uri string) error {
 	} else {
 		return fmt.Errorf("the URI provided: %s must begin with http/https", uri)
 	}
+	// The URL will contain a checksum after the '?' so we can just check for the suffix followed by a '?'.
+	if !strings.HasSuffix(parsedURL.Path, "qcow2.gz") && !strings.HasSuffix(parsedURL.Path, "qcow2.xz") {
```

Comment:
I think you want to ensure that there is the `.` before `qcows`.
```suggestion
	if !strings.HasSuffix(parsedURL.Path, ".qcow2.gz") && !strings.HasSuffix(parsedURL.Path, ".qcow2.xz") {
```

Reply:
yep. fixed